.rubocop.yml

```diff
@@ -18,6 +18,12 @@ AllCops:
     - 'truffleruby/target/classes/**/*'
     - 'mxbuild/**/*'
 
+Lint/AssignmentInCondition:
+  Description: Don't use assignment in conditions.
+  StyleGuide: "#safe-assignment-in-condition"
+  Enabled: true
+  AllowSafeAssignment: true
+
```

Comment:
Just to clarify, because I'm really arguing two different things here:

1) I think assignments in conditionals are problematic. I've been writing Ruby professionally for 9 years now and while you might not screw them up frequently, when you do they can be a gigantic waste of time to track down. This is no different than other languages with the same construct, other than the rules of truthiness being different. I don't think our code base is immune to this issue. @eregon and I obviously view this very differently, so it's probably best to just set that aside.

2) To the extent that I would like the author's intent of `=` vs `==` to be clearly spelled out, it would be nice if there were a language construct or a RuboCop rule to do it. However, I've been convinced that `()` is not a great way of doing it. It can just as easily be misread and has valid use as a grouping operator, at which point we're right back where we started. So, I'm in favor of removing the rule.